File: MVC y DAO/src/main/java/co/edu/unisalle/crud/controller/EstudianteController.java
package co.edu.unisalle.crud.controller;

import co.edu.unisalle.crud.dao.IEstudianteDAO;
import co.edu.unisalle.crud.model.Estudiante;

import java.util.List;
import java.util.Optional;

/**
 * Controlador del patrón MVC.
 *
 * SRP: media entre la vista y el DAO sin contener reglas de persistencia ni
 * lógica de presentación.
 * DIP: depende de la abstracción {@link IEstudianteDAO}, no de su
 * implementación concreta — la implementación se inyecta por constructor.
 */
public class EstudianteController {

    private final IEstudianteDAO dao;

    public EstudianteController(IEstudianteDAO dao) {
        this.dao = dao;
    }

    public Estudiante crear(int id, String nombre, String apellido, String correo, String programa) {
        Estudiante estudiante = new Estudiante(id, nombre, apellido, correo, programa);
        return dao.crear(estudiante);
    }

    public List<Estudiante> listar() {
        return dao.listar();
    }

    public Optional<Estudiante> buscarPorId(int id) {
        return dao.buscarPorId(id);
    }

    public Estudiante actualizar(int id, String nombre, String apellido, String correo, String programa) {
        Estudiante estudiante = new Estudiante(id, nombre, apellido, correo, programa);
        return dao.actualizar(estudiante);
    }

    public boolean eliminar(int id) {
        return dao.eliminar(id);
    }
}

File: MVC y DAO/CrudMvcDao/Controller/EstudianteController.cs
using CrudMvcDao.Dao;
using CrudMvcDao.Model;

namespace CrudMvcDao.Controller;

/// <summary>
/// Controlador del patrón MVC.
///
/// SRP: media entre la vista y el DAO sin contener reglas de persistencia ni
/// lógica de presentación.
/// DIP: depende de la abstracción <see cref="IEstudianteDao"/>, no de su
/// implementación concreta — la implementación se inyecta por constructor.
/// </summary>
public class EstudianteController
{
    private readonly IEstudianteDao _dao;

    public EstudianteController(IEstudianteDao dao)
    {
        _dao = dao;
    }

    public Estudiante Crear(int id, string nombre, string apellido, string correo, string programa) =>
        _dao.Crear(new Estudiante(id, nombre, apellido, correo, programa));

    public IReadOnlyList<Estudiante> Listar() => _dao.Listar();

    public Estudiante? BuscarPorId(int id) => _dao.BuscarPorId(id);

    public Estudiante Actualizar(int id, string nombre, string apellido, string correo, string programa) =>
        _dao.Actualizar(new Estudiante(id, nombre, apellido, correo, programa));

    public bool Eliminar(int id) => _dao.Eliminar(id);
}
